src/icf/utils/struct.ts

```diff
@@ -0,0 +1,129 @@
+import { AbiType, AbiItemType, StructAbiItemType } from '../abiTypes';
+import { castStatement, reverseCastStatement, transformType } from './util';
+import { INDENT, getInteractiveFuncs } from '../genCairo';
+
+function copyStructItem(struct: StructAbiItemType): StructAbiItemType {
+  return {
+    type: 'struct',
+    name: struct.name,
+    size: struct.size,
+    members: struct.members.map((member) => {
+      return { name: member.name, offset: member.offset, type: member.type };
+    }),
+  };
+}
+
+export function getStructsFromABI(abi: AbiType): StructAbiItemType[] {
+  const result: StructAbiItemType[] = [];
+  abi.forEach((item) => {
```

Comment:
you can replace this forEach with filter

Reply:
We will have to use type-guards
 https://stackoverflow.com/questions/43010737/way-to-tell-typescript-compiler-array-prototype-filter-removes-certain-types-fro